### ISSUE TITLE
Filer log cleanup?

### ISSUE THREAD

**Issue by divanikus** [seaweedfs/seaweedfs#3455]
**Describe the bug**
What is the proper way of cleaning filer log? Most of my buckets are with 1-2w TTL. So data is rotated very quickly. I end up with really lenghty filer log, those collectionless volumes.

I've tried to clean it manually, by removing previous months from `/topics/.system/log/`. But as far as I understand, `volume.vacuum` doesn't really vacuum them.

For example I see this on one of my volume servers. 
![image](https://user-images.githubusercontent.com/3438036/184981264-03ec0c62-7ec1-4e91-9d87-a0e509c6234d.png)

Trying to issue `volume.vacuum -volumeId 67454` does nothing. So now I have hundreds of those "collectionless" volumes which I don't know how to get rid of and reclaim some space.

**System Setup**
- OS version: Debian 11
- output of `weed version`: 
```
version 30GB 3.18 475185fb723b50e05f0afe39f7f529302bcee11a linux amd64
```

**Comment by grepler** [comment]
Chiming in, now that running sqlite is *technically* feasible on a FUSE mounted volume, I gave it a shot - the filer changelogs rapidly built up space because of the sqlite activity.

and as far as I understand it, once we remove any historical folders from the `/topics/.system/log/` dir, filter sync can no longer bootstrap up a new filer, is that right?

**Comment by chrislusf** [comment]
> and as far as I understand it, once we remove any historical folders from the `/topics/.system/log/` dir, filer sync can no longer bootstrap up a new filer, is that right?

Correct. 

There are plans to compact the log history. @lyang24

**Comment by alzamonn** [comment]
@chrislusf hi.
Could you give some advice please about clean up huge of logs collecting here /topics/.system/log/ (v.3.29)
1. Is any way to clear(rotate) those logs? 
2. Last version can handled it?
```
[3.37](https://github.com/seaweedfs/seaweedfs/releases/tag/3.37) [Latest]
What's Changed
- Rotate log files.
- Mount
......
```

**Comment by chrislusf** [comment]
you can delete the old ones

**Comment by alzamonn** [comment]
@chrislusf thank you. I cleaned up today some of those logs.
 For instance I had such values size of logs:
> fs.du /topics/.system/log/2022-12-15
block:6644      logical size:**22GB**       /topics/.system/log/2022-12-15

> fs.du /topics/.system/log/2022-12-16
block:19167     logical size:**67GB**       /topics/.system/log/2022-12-16
In the same time the data which stored on seaweedfs took only 4GB. Periodically there is file rotation when updating this data. 

Please tell, are you planning add some rotate/fix size of logs stored here /topics/.system/log/ ?

**Comment by chrislusf** [comment]
the log size is proportional to the number of file operations, not the size of the files.

**Comment by alzamonn** [comment]
@chrislusf yes, on the same date above, we had lot of operation with files

Sorry for repeating. Have you planning add some rotate/fix size of logs stored here /topics/.system/log/ ?
Thank you for your reply

**Comment by grepler** [comment]
> you can delete the old ones

Note thought that this will (IIUC) break filer replication - you will not be able to spin up a new filer and 'replay' the filesystem events to the remote site.
@chrislusf is there a way to set up filer replication after the logs dir has been truncated?

**Comment by chrislusf** [comment]
The log replaying would not be enough to replicate the filer store. In the future, it needs to start from a snapshot.

**Comment by Dando-Real-ITA** [comment]
Any news on the logs?
I currently use seaweedfs as a distributed memory cache, and it happens that volumes disappear, then mounts can't sync anymore correctly
Would be possible a filer setting to have like a timed auto cleanup of the topics folder ( like 30d older topics are deleted )?

**Comment by Dando-Real-ITA** [comment]
Update on the topic, debugging a space usage problem I found that logs are taking up most of the file system, a 10 GB filesystem is taking up 2.5 GB of logs for every day.
I for now clean up manually these logs, but a compact and/or an integrated clean up solution would be really usefull

**Comment by Dando-Real-ITA** [comment]
How much deleting logs impacts starting a new FUSE mount? Would it be able to use the current metadata ( i.e. the same generated from fs.meta.save ) as a starting point?
Same question for a new filer bootstrap

**Comment by sgalsaleh** [comment]
@chrislusf @divanikus 

Should the following procedure to expunge logs while maintaining the metadata work? And would that have any effect on filer replication?

- Scale down all filers
- Purge all filers logs
- Scale up all filers

**Comment by chrislusf** [comment]
```
> fs.log.purge -h
Usage of fs.log.purge:
  -daysAgo uint
    	purge logs older than N days (default 365)
  -v	verbose mode
```

**Comment by sgalsaleh** [comment]
Would using `fs.log.purge` affect the ability to add more filers later?

**Comment by chrislusf** [comment]
> Would using `fs.log.purge` affect the ability to add more filers later?

yes. this area needs some work to export the existing filer metadata and load them.

**Comment by zemul** [comment]
I had the same problem.

Which plan is better? @chrislusf 
1. mount Add startup parameters to disable writing logs
2. For frequent operations like sqlLite, implement 4k blocks to 2m blocks and merge

**Comment by xpader** [comment]
`topics/.system/log/` is disgusting.

**Comment by xpader** [comment]
We use loki + seaweedfs s3 to storage loki's chunk file, because seaweedfs is good solution to storage massive amounts of small files, and loki use lots of small file to storage chunks.

I think it's perfect match before, but when write lots of logs to loki all the day time, seaweedfs will generate lots of logs too. Over over the time, seaweedfs logs become more than loki chunks, this became a hurge burden.

If there is an option to disable logs, or an option to auto rotate logs everyday, will be awsome.
